src/encoded/static/components/matrix_target.js

```diff
@@ -0,0 +1,677 @@
+        const type = pluralize(queryStringType.toLocaleLowerCase());
+        return (
+            <div className="matrix-general-search">
+                <p>Enter search terms to filter the {type} included in the target matrix.</p>
+                <div className="general-search-entry">
+                    <i className="icon icon-search" />
+                    <div className="searchform">
+                        <TargetMatrixTextFilter filters={context.filters} searchBase={matrixSearch} onChange={this.onChange} />
+                    </div>
+                </div>
+            </div>
+        );
+    }
+}
+
+/**
+ * Render the area above the matrix itself, including the page title.
+ */
+class TargetMatrixHeader extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            context: this.props.context,
+            searchResultCount: this.props.context.total,
+        };
+        this.setSearchResultCount = this.setSearchResultCount.bind(this);
+    }
+
+    componentDidMount() {
+        this.searchCount = PubSub.subscribe(SEARCH_RESULT_COUNT_PUBSUB, this.setSearchResultCount);
+    }
+
+    componentWillUnmount() {
+        PubSub.unsubscribe(this.searchCount);
+    }
+
+    setSearchResultCount(message, searchResultCount) {
+        this.setState({ searchResultCount });
+    }
+
+    render() {
+        const visualizeDisabledTitle = this.state.context.total > VISUALIZE_LIMIT ? `Filter to ${VISUALIZE_LIMIT} to visualize` : '';
+
+        return (
+            <div className="matrix-header">
+                <div className="matrix-header__title">
+                    <h1>{this.state.context.title}</h1>
+                    <div className="matrix-tags">
+                        <MatrixInternalTags context={this.state.context} />
+                    </div>
+                </div>
+                <div className="matrix-header__controls">
+                    <div className="matrix-header__target-filter-controls">
+                        <TargetMatrixSearch context={this.state.context} />
+                    </div>
+                    <div className="matrix-header__target-search-controls">
+                        <h4>Showing {this.state.searchResultCount} results</h4>
+                        <SearchControls context={this.state.context} visualizeDisabledTitle={visualizeDisabledTitle} hideBrowserSelector />
+                    </div>
+                </div>
+            </div>
+        );
+    }
+}
+
+
+TargetMatrixHeader.propTypes = {
+    /** Matrix search result object */
+    context: PropTypes.object.isRequired,
+};
+
+const TargetMatrixContent = ({ context }) => (
+    <div className="matrix__content matrix__content--reference-epigenome">
+        <TargetMatrixPresentation context={context} />
+    </div>
+);
+
+TargetMatrixContent.propTypes = {
+    /** Matrix search result object */
+    context: PropTypes.object.isRequired,
+};
+
+
+class TargetTabPanel extends TabPanel {
```

Comment:
I needed fine-grain control to experiment, so I extend rather than use TabPanel. Like DataTable, I am open to trying to use TabPanel with modifications if you feel that is needed.

Reply:
I think you’ll have to use TabPanel — again the warning against React component inheritance. I think we’ll have to go this direction, especially as we phase out component classes.